File: backend/GraduateProject.Application/RealEstate/UserAccountDto/UserAccount.ListDto.cs
using System.Collections.Generic;
using Microsoft.AspNetCore.Identity;
using GraduateProject.Domain.Ums.Entities;
namespace GraduateProject.Application.RealEstate.UserAccountDto;

    public class UserAccountListDto
    {  
         public string? FullName { get; set; }
         public string? Email { get; set; }
         public bool? Active { get; set; } = true;
         public string? UserName { get; set; }
         public string? PhoneNumber { get; set; }
         // public string? RoleId { get; set; }
         // public string? Code { get; set; }
         public ICollection<UserRole> UserRoles { get; set; } = new List<UserRole>();
         //public ICollection<Post> Posts { get; set; } = new List<Post>();
         //public IEnumerable<Role> GetRoles() => UserRoles.Select(x => x.Role).ToList();

    }

File: frontend/src/types/UserAccountListDto.ts
import { Identity } from "./Common";
import { Role } from "./Role";

interface UserRole {
	roleId: string;
	role: Role;
}

export type UserAccountListDto = {
	fullName: string;
	email: string;
	active: string;
	userName: string;
	phoneNumber: string;
	userRoles: UserRole[];
};

export type UserAccountTable = {
	fullName: string;
	email: string;
	active: string;
	userName: string;
	phoneNumber: string;
	roleId: string;
	code: string;
};
